public/components/FeatureAnywhereContextMenu/CreateAnomalyDetector/AssociateExisting/containers/AssociateExisting.tsx

```diff
@@ -0,0 +1,282 @@
+
+    // Map all detector IDs for all the found augmented vis objects
+    const savedAugmentDetectorsSet = new Set(
+      savedAugmentForThisVisualization.map((savedObject) =>
+        get(savedObject, 'pluginResourceId', '')
+      )
+    );
+
+    // detectors here is all detectors
+    // for each detector in all detectors return that detector if that detector ID isnt in the set
+    // filter out any detectors that aren't on the set of detectors IDs from the augmented vis objects.
+    const detectorsToDisplay = detectors.filter((detector) => {
+      if (
+        !savedAugmentDetectorsSet.has(detector.id) &&
+        detector.detectorType === 'SINGLE_ENTITY'
+      ) {
+        return detector;
+      }
+    });
+    return detectorsToDisplay;
+  };
+
+  useEffect(() => {
+    getDetectors();
+  }, []);
+
+  const getDetectors = async () => {
+    dispatch(getDetectorList(GET_ALL_DETECTORS_QUERY_PARAMS));
+  };
+
+  const selectedOptions = useMemo(() => {
+    if (
+      !existingDetectorsAvailableToAssociate ||
+      !associateExistingProps.selectedDetector
+    ) {
+      return [];
+    }
+
+    const detector = (existingDetectorsAvailableToAssociate || []).find(
+      (detector) =>
+        detector.id === get(associateExistingProps.selectedDetector, 'id', '')
+    );
+    return detector ? [{ label: detector.name }] : [];
+  }, [
+    associateExistingProps.selectedDetector,
+    existingDetectorsAvailableToAssociate,
+  ]);
+
+  const detector = useMemo(
+    () =>
+      existingDetectorsAvailableToAssociate &&
+      associateExistingProps.selectedDetector &&
+      existingDetectorsAvailableToAssociate.find(
+        (detector) =>
+          detector.id === get(associateExistingProps.selectedDetector, 'id', '')
+      ),
+    [
+      associateExistingProps.selectedDetector,
+      existingDetectorsAvailableToAssociate,
+    ]
+  );
+  const options = useMemo(() => {
+    if (!existingDetectorsAvailableToAssociate) {
+      return [];
+    }
+
+    return existingDetectorsAvailableToAssociate.map((detector) => ({
+      label: detector.name,
+    }));
+  }, [existingDetectorsAvailableToAssociate]);
+
+  return (
+    <div className="associate-existing">
+      <EuiText size="xs">
+        <p>
+          View existing anomaly detectors across your system and add the
+          detector(s) to a dashboard and visualization.{' '}
+          <a href={`${BASE_DOCS_LINK}/ad`} target="_blank">
+            Learn more <EuiIcon type="popout" />
+          </a>
+        </p>
+      </EuiText>
+      <EuiSpacer size="l" />
+      <EuiTitle size="s">
+        <h3>Select detector to associate</h3>
+      </EuiTitle>
+      <EuiSpacer size="m" />
+      <EuiText size="xs" color="subdued" style={{ paddingBottom: '3px' }}>
+        Eligible detectors don't include high-cardinality detectors.
+      </EuiText>
+      {!existingDetectorsAvailableToAssociate && <EuiLoadingSpinner size="l" />}
+      {existingDetectorsAvailableToAssociate && (
```

Comment:
nit: this could be simplified to a single conditional instead of separate entries for each scenario. Example:
```
{ existingDetectorsAvailableToAssociate ?
  <EuiComboBox ...>
 : 
 <EuiLoadingSpinner ...>
```

I think this makes it a little cleaner, especially since these are both exclusive to each other and will never be rendered together.